Sources/Fuzzilli/Evaluation/ProgramCoverageEvaluator.swift

```diff
@@ -15,29 +15,72 @@
 import Foundation
 import libcoverage
 
-class CovEdgeSet: ProgramAspects {
-    let count: UInt64
-    let edges: UnsafeMutablePointer<UInt32>?
+public class CovEdgeSet: ProgramAspects {
+    var count: UInt64
+    var edges: UnsafeMutablePointer<UInt32>?
 
     init(edges: UnsafeMutablePointer<UInt32>?, count: UInt64) {
         self.count = count
         self.edges = edges
         super.init(outcome: .succeeded)
     }
-    
+
     deinit {
         free(edges)
     }
 
-    override var description: String {
+    public override var description: String {
         return "new coverage: \(count) newly discovered edge\(count > 1 ? "s" : "") in the CFG of the target"
     }
+
+    /// This adds additional copies, but is only hit when new programs are added to the corpus
+    /// It is used by corpus schedulers such as MarkovCorpus that require knowledge of which samples trigger which edges
+    public func toEdges() -> [UInt32] {
+        return Array(UnsafeBufferPointer(start: edges, count: Int(count)))
+    }
+
+    public static func == (lhs: CovEdgeSet, rhs: CovEdgeSet) -> Bool {
+        if lhs.outcome != rhs.outcome { return false }
+        if lhs.count != rhs.count { return false }
+        for i in 0..<Int(lhs.count) {
+            if lhs.edges![i] != rhs.edges![i] {
+                return false
+            }
+        }
+        return true
+    }
+
+    public override func intersect(_ otherAspect: ProgramAspects) -> Bool {
```

Comment:
Nit: maybe make it `formIntersection(_ otherAspects: ProgramAspects)` to follow similar Swift methods in other classes

Also, what does the return value indicate?

Reply:
In the updated version, the boolean indicates whether or not the intersection is smaller than the initial aspect provided. This is used to determine when determinism has been reached.

An alternative would be to build a means of comparing the size of `ProgramAspects`